### PR TITLE
feat:[ADL-P] Add MaxLed support for send POST code over I2C

### PR DESCRIPTION
**Pull request by pastorcx**
- add MaxLed Module support for Max6950
- send POST code over I2C

### CODE REVIEW

File: Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.h [hunk 1]
```diff
@@ -107,4 +107,15 @@ RtcInit (
...
   );
 
+/**
+  This function does the basic init work for Max6950.
+
+  @param[in]  SpiBaseAddress - Base Address of SPI Controller
+**/
+EFI_STATUS
+EFIAPI
+MaxLedInit (
```
Comment (gdong1) [inline Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.h:117]: This is a module internal function, EFIAPI is not required.

File: Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/MaxLedInit.c [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      &I2cNumber,
+      SlaveAddress,
+      2,
+      WriBuf,
+      I2C_BUS_TIMEOUT_USEC,
+      I2C_BUS_FREQUENCY_400Khz,
+      NULL,
+      TRUE,
+      TRUE
+  );
```
Comment (gdong1) [inline Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/MaxLedInit.c:157]: Following coding style, subsequent lines of multi-line function calls should line up 2 spaces from the beginning of the function name. It looks like:
Status = SerialIoI2cWrite (
                PciCfgBase,
                &I2cNumber,
                ...,
              );

File: Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/MaxLedInit.c [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+#include <Library/PciLib.h>
+#include <Library/PchInfoLib.h>
+#include <Library/TimerLib.h>
+#include <PchAccess.h>
+#include <Register/PchRegsLpc.h>
+#include <Register/PchRegsPcr.h>
+#include <Library/PchPciBdfLib.h>
+#include <PlatformBoardId.h>
+#include <Library/SmbusLib.h>
+#include <Library/SerialIoI2cLib.h>
```
Comment (gdong1) [inline Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/MaxLedInit.c:23]: It looks some header files are not really used. better only include necessary files.

File: Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.h (hunk 1 above)
Comment (pastorcx) [reply Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.h:116]: done

File: Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/MaxLedInit.c (hunk 2 above)
Comment (pastorcx) [reply Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/MaxLedInit.c:145]: updated coding style, done

File: Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/MaxLedInit.c (hunk 3 above)
Comment (pastorcx) [reply Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/MaxLedInit.c:11]: removed header files that are not needed

File: Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/MaxLedInit.c (hunk 2 above)
Comment (gdong1) [reply Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/MaxLedInit.c:145]: The comment mentioned "line up 2 spaces from the beginning of the function name. ", but the example doesn't show it as expected even I did it in the comment. you could refer   https://github.com/slimbootloader/slimbootloader/blob/45273c82ca7ab9433468d8afc456856ee478ee1f/MdePkg/Library/BasePciCf8Lib/PciCf8Lib.c#L567

Comment (pastorcx) [reply Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/MaxLedInit.c]: updated as in the given example

**Review by gdong1 — APPROVED** [review]

**Pull request merged**